src/work-allocation-2/components/task-list-filter/task-list-filter.component.ts

```diff
@@ -41,20 +45,32 @@ export class TaskListFilterComponent implements OnInit, OnDestroy {
   };
   public selectedLocations: string[] = [];
   public toggleFilter = false;
-
+  public errorSubscription: Subscription;
+  public appStoreSub: Subscription;
   private locationSubscription: Subscription;
   private selectedLocationsSubscription: Subscription;
-  public errorSubscription: Subscription;
 
   /**
    * Accept the SessionStorageService for adding to and retrieving from sessionStorage.
    */
   constructor(private readonly route: ActivatedRoute,
+              private readonly store: Store<fromRoot.State>,
               private readonly filterService: FilterService,
               private readonly locationService: LocationDataService) {
   }
 
   public ngOnInit(): void {
+
+    this.appStoreSub = this.store.pipe(select(fromRoot.getUserDetails)).subscribe(
```

Comment:
The logic to use which persistence should not be part of this component. 

- We want it as an input parameter.
- The logic should be moved to an Util method.
- Lastly, the container component should call the util method and pass it to this component as Input